common/utils/backend.py

```diff
@@ -357,6 +358,7 @@ def get_next_num_run(self, peek: bool = False) -> int:
         other_num_runs = [
             int(os.path.basename(run_dir).split("_")[1])
             for run_dir in glob.glob(os.path.join(self.internals_directory, "runs", "*"))
+            if re.match(r"\d+_\d+_\d+", os.path.basename(run_dir))
```

Comment:
Do you think we can either:
1- Add a comment explaining that this is <seed>_<num_run>_<budget>
2- I think budget is a float, so it might confuse someone. So should we create a function that is called is_run_dir that explains this in the docstring?

Functionally I do not see any problem. I am thinking if there is a better way to prevent a problem in case someone adds an extra '\d' In the future :)

Reply:
we can make a string pattern that we use while storing num_run_directory [here](https://github.com/automl/automl_common/blob/93984e17464a2e9078d3b415528f06784515001e/common/utils/backend.py#L330). And translate this into the regex pattern in the is_run_dir function.